docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,347 @@
+= Tutorial - Adding a new Command
+:toc: macro
+:site-section: DeveloperGuide
+:imagesDir: ../images/add-remark
+:stylesDir: ../stylesheets
+:xrefstyle: full
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:warning-caption: :warning:
+endif::[]
+
+toc::[]
+
+== Introduction
+
+In this tutorial, we'll walk you through the implementation of a new command -- `remark`.
+
+This command allows users of the AddressBook application to add optional remarks to people in their address book and edit it if required.
+The command should have the format of `remark INDEX r/REMARK`.
+An example of the command is `remark 2 r/Likes baseball`.
+
+We'll assume that you have already set up the development environment as outlined in the Developer's Guide.
+
+=== Create a new `Command`
+
+Looking in the `logic.command` package, you will notice that each existing command have their own class.
+All the commands implement `Command` which means that they must override `execute()`.
+Each `Command` returns an instance of `CommandResult` upon success and throws a `CommandException` otherwise.
+This exception is handled gracefully -- the error message provided to `CommandException` is shown in the `ResultDisplay`.
+
+.UML diagram describing the relationship between RemarkCommand and Command
```

Comment:
UML diagram describing the relationship between RemarkCommand and Command -->
~~UML diagram describing~~ The relationship between `RemarkCommand` and `Command`

There are similar cases in other places.